nixos/modules/system/boot/loader/raspberrypi/builder_uboot.nix

```diff
@@ -0,0 +1,58 @@
+{ config, pkgs }:
+
+let
+  is64bit = pkgs.stdenv.is64bit;
+  optional = pkgs.stdenv.lib.optionalString;
+
+  cfg = config.boot.loader.raspberryPi;
+
+  uboot =
+    if cfg.version == 2 then
+      if is64bit then
+        throw ("Raspberry Pi 2 does not support 64bit")
+      else
+        pkgs.ubootRaspberryPi2
+    else if cfg.version == 3 then
+      if is64bit then
+        pkgs.ubootRaspberryPi3_64bit
+      else
+        pkgs.ubootRaspberryPi3_32bit
+    else
+      throw ("raspberrypi-uboot: unknown version: " ++ cfg.version);
```

Comment:
In NixOS configurations, throw shouldn't be used to validate stuff but the `assertions` option.